server/verify.go

```diff
@@ -175,44 +187,44 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 
 // validateAK validates AK cert in the attestation, and returns AK cert (if exists) and public key.
 // It also pulls out the GCE Instance Info if it exists.
-func validateAK(attestation *pb.Attestation, opts VerifyOpts) (*pb.MachineState, crypto.PublicKey, error) {
+func validateAK(attestation *pb.Attestation, opts VerifyOpts) (*pb.MachineState, tpm2.Public, crypto.PublicKey, error) {
+	// If the AK Cert is not in the attestation, use the AK Public Area.
+	akPubArea, err := tpm2.DecodePublic(attestation.GetAkPub())
+	if err != nil {
+		return nil, tpm2.Public{}, nil, fmt.Errorf("failed to decode AK public area: %w", err)
+	}
 	if len(attestation.GetAkCert()) == 0 || len(opts.TrustedRootCerts) == 0 {
-		// If the AK Cert is not in the attestation, use the AK Public Area.
-		akPubArea, err := tpm2.DecodePublic(attestation.GetAkPub())
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to decode AK public area: %w", err)
-		}
 		akPubKey, err := akPubArea.Key()
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get AK public key: %w", err)
+			return nil, tpm2.Public{}, nil, fmt.Errorf("failed to get AK public key: %w", err)
 		}
 		if err := validateAKPub(akPubKey, opts); err != nil {
```

Comment:
Also , since we are using the "content" in the ak, we better `validateAKPub` before using its "content", this check is currently skipped if any ak cert presents